src/instructlab/train/pytorch_train.py

```diff
@@ -193,40 +201,53 @@ def linux_train(
     )
 
     if four_bit_quant:
-        print("LINUX_TRAIN.PY: USING 4-bit quantization with BitsAndBytes")
-        use_fp16 = True
+        print("PYTORCH_TRAIN.PY: USING 4-bit quantization with BitsAndBytes")
         bnb_config = BitsAndBytesConfig(
             load_in_4bit=True,
             bnb_4bit_quant_type="nf4",
             bnb_4bit_use_double_quant=True,
             bnb_4bit_compute_dtype=torch.float16,  # if not set will throw a warning about slow speeds when training
         )
     else:
-        print("LINUX_TRAIN.PY: NOT USING 4-bit quantization")
-        use_fp16 = False
+        print("PYTORCH_TRAIN.PY: NOT USING 4-bit quantization")
         bnb_config = None
 
     # Loading the model
-    print("LINUX_TRAIN.PY: LOADING THE BASE MODEL")
+    print("PYTORCH_TRAIN.PY: LOADING THE BASE MODEL")
     config = AutoConfig.from_pretrained(
         model_name, torchscript=True, trust_remote_code=True
     )
 
+    torchDtype = None
+    if dtype == "fp16":
+        torchDtype = torch.float16
+    if dtype == "bf16":
+        torchDtype = torch.bfloat16
+    if dtype == "fp32":
+        torchDtype = torch.float32
+    else:
+        torchDtype = "auto"
```

Comment:
This PR cleans up linux train as well, in that dtype was hardcoded depending on GPU type. Adding another gpu type would lead to another hardcoded situation for the dtype. Bubbling this up as an option seems to be the right move as we add support for more gpu backends.